api/v1alpha1/controlplane_types.go

```diff
@@ -71,16 +79,24 @@ const (
 	BackendDBTypeDedicated BackendDBType = "dedicated"
 )
 
-// +kubebuilder:validation:Enum=k8s;ocm;vcluster;host
+// +kubebuilder:validation:Enum=k8s;ocm;vcluster;host;external
 type ControlPlaneType string
 
 const (
 	ControlPlaneTypeK8S      ControlPlaneType = "k8s"
 	ControlPlaneTypeOCM      ControlPlaneType = "ocm"
 	ControlPlaneTypeVCluster ControlPlaneType = "vcluster"
 	ControlPlaneTypeHost     ControlPlaneType = "host"
+	ControlPlaneTypeExternal ControlPlaneType = "external"
 )
 
+// SecretReference is a reference to a secret that holds the kubeconfigs for
+// a control plane hosted in the kubeflex hosting cluster, or for a kubeconfig
+// for the hosting cluster itself (in the case of a control plane of type 'host').
```

Comment:
This wording does not seem to cover "external" type control planes.

Reply:
Updated to reflect that it can be used for external types control planes.